Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/AuthController.kt

```diff
@@ -39,15 +50,27 @@ class AuthController(@DrawableRes private val loginScreenAppIcon: Int) {
 
         return if (resultCode == Activity.RESULT_OK) {
             user = FirebaseAuth.getInstance().currentUser
+            fetchDataFromFirebase()
             null
         } else {
             response?.error?.message
         }
     }
 
     fun logout(context: Context) {
-        AuthUI.getInstance()
-            .signOut(context)
+        AuthUI.getInstance().signOut(context)
         user = null
     }
+
+    override fun onDataChange(dataSnapshot: DataSnapshot) {
+        if (user == null) {
+            //TODO clear the user cached in the Firebase database.
+        } else {
+            //TODO store an instance of the user in the Firebase database for use throughout app.
+        }
+    }
+
+    override fun onCancelled(databaseError: DatabaseError) {
+        Log.e(javaClass.simpleName, databaseError.message, databaseError.toException())
```

Comment:
Do we want to create an error code or a standard for logging errors or we are good like this?

cc @n8ebel @ericmaxwell2003

Reply:
Consistency is a good idea.  Are these errors we might recover from? Or are these specific types of database errors we may want to track?  

I tend to use Timber for logging.

Reply:
Since the firebase database is executing async, it could emit an error after the SSO/Firebase Auth network call is successful. We could pop a dialog to the user as well as Log it to the console, similar to how the MainActivity class is doing it onActivityResult if Firebase Auth fails.

For logging, it seems Log.e() was used elsewhere in the app which is why I used it here.

Reply:
Thanks @CCorrado for diving deep on this, do we want to chat over slack about the logging strategy?

Reply:
sure. I ended up removing this code but i think should still align on one.

Reply:
2nd Timber...super trivial to add a Timber tree and Crashlytics tree.